index.js

```diff
@@ -1,13 +1,16 @@
 'use strict';
 
-var mkdirp = require('mkdirp');
-var md5Hex = require('md5-hex');
 var fs = require('fs');
 var path = require('path');
+var mkdirp = require('mkdirp');
+var md5Hex = require('md5-hex');
 var writeFileAtomic = require('write-file-atomic');
+var packageHash = require('package-hash');
```

Comment:
Hopefully the hash is cached though, so `require.resolve` only needs to be used once per package.

`git` won't be called unless `.git/HEAD` exists, so that should really only affect symlinked repositories. For regular dependencies there's:

* one recommended `require.resolve` call
* one `statSync`
* one `readFileSync` for `package.json`
* one `readFileSync` for `.git/HEAD`, which will fail

I can't quite find how Node's module loading works but I doubt this is many more operations than `require('foo/package.json').version`.